File: file-management/Models/Domains/Document.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace file_management.Models.Domains
{
    public class Document
    {
        public string DocumentId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Path { get; set; }
        public string Extension { get; set; }
        public long SizeInBytes { get; set; }
        public DateTime ActivedDate { get; set; }

        public Guid DocumentCategoryId { get; set; }
        public DocumentCategory DocumentCategory { get; set; }
    }
}

File: file-management-ui/src/interfaces/document.ts
export interface IDocumentForm {
	name: string,
	description: string,
	documentCategoryId: string,
}

export interface IDocumentCategory {
	documentCategoryId: string,
	name: string,
	description: string
}

export interface IDocument {
	documentId: string,
	name: string,
	description: string,
	path?: string,
	extension?: string
	sizeInBytes: number,
	activedDate: string,
	documentCategoryId:string,
	documentCategory?: IDocumentCategory,
	apiBaseUrl?: string
}

/*************************************************
 * 	Document Category Response Part
 */
export interface IDocumentCategoryDataResponse {
	total: number,
	pageNo: number,
	pageSize: number,
	apiBaseUrl?: string,
	results: IDocumentCategory[]
}

export interface IDocumentCategoryResponse {
	isSuccess: boolean,
	statusCode: number,
	errorMessage: Array<string>,
	data: IDocumentCategoryDataResponse
}

/*************************************************
 * 	Documents Response Part
 */
export interface IDocumentDataResponse {
	total: number,
	pageNo: number,
	pageSize: number,
	apiBaseUrl?: string,
	results: IDocument[]
}

export interface ISingleDocumentResponse {
	isSuccess: boolean,
	statusCode: number,
	errorMessage: Array<string>,
	data: IDocument
}

export interface IDocumentResponse {
	isSuccess: boolean,
	statusCode: number,
	errorMessage: Array<string>,
	data: IDocumentDataResponse
}